SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -219,6 +221,47 @@ void InterestFactory::AddServerSelfInterest(Interest& OutInterest, const Worker_
 	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::ENTITY_ACL_COMPONENT_ID, LoadBalanceQuery);
 }
 
+bool InterestFactory::CheckOwnersHaveEntityId(const AActor* Actor) const
+{
+	AActor* Owner = Actor->GetOwner();
+
+	while (Owner != nullptr && !Owner->IsPendingKillPending())
+	{
+		if (PackageMap->GetEntityIdFromObject(Owner) == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			return false;
+		}
+		Owner = Owner->GetOwner();
+	}
+
+	return true;
+}
+
+void InterestFactory::AddOwnerInterestOnServer(Interest& OutInterest, const AActor* InActor) const
+{
+	AActor* Owner = InActor->GetOwner();
+	Query OwnerChainQuery;
+	
+	while (Owner != nullptr && !Owner->IsPendingKillPending())
+	{
+		QueryConstraint OwnerQuery;
+		OwnerQuery.EntityIdConstraint = PackageMap->GetEntityIdFromObject(Owner);
+		if (OwnerQuery.EntityIdConstraint == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			return ;
+		}
+		OwnerChainQuery.Constraint.OrConstraint.Add(OwnerQuery);
+		Owner = Owner->GetOwner();
+	}
+
+	if (OwnerChainQuery.Constraint.OrConstraint.Num() != 0)
+	{
+		OwnerChainQuery.ResultComponentIds = ServerNonAuthInterestResultType;
+		OwnerChainQuery.ResultComponentIds.Add(SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
```

Comment:
Should this just be included in the NonAuth interest?

Reply:
I don't think this is different to nonauth interest - so not sure we needed a different interest type